cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -354,3 +360,13 @@ type AuditPolicyConfiguration struct {
 	LogMaxAge *int32 `json:"logMaxAge,omitempty"`
 	//TODO(chuckha) add other options for audit policy.
 }
+
+// KubeadmTimeouts indicates timeouts managed by kubeadm.
+type KubeadmTimeout string
+
+const (
+	// WaitForControlPlaneTimeout indicates the timeout duration kubeadm should wait when waiting for the control-plane
+	// to start in the kubeadm init workflow and in the kubeadm join controlplane workflow
+	// defaults to 4 minutes
+	WaitForControlPlaneTimeout = "controlplane"
```

Comment:
a question here would be, what other types of timeouts we want to add for the init and join configurations?

Reply:
@neolit123 possibly discovery timeout, although I highly recommend keeping the original spot it occupies (`JoinConfiguration.Discovery.Timeout`)